pkg/dsdk/remote_provider.go

```diff
@@ -192,7 +193,13 @@ type RemoteProviderDeleteRequest struct {
 }
 
 func (e *RemoteProvider) Delete(ro *RemoteProviderDeleteRequest) (*RemoteProvider, *ApiErrorResponse, error) {
-	rs, apierr, err := GetConn(ro.Ctxt).Delete(ro.Ctxt, e.Path, nil)
+	v := reflect.ValueOf(*ro)
```

Comment:
This is the place where v can get zero value and is passed to formatQueryParams() without validation. The formatQueryParams() is also not validating it and hence v.Field(i).Interface() would fail. A nil validation is needed at some place.